src/app.rs

```diff
@@ -95,14 +88,24 @@ impl App {
     #[allow(clippy::missing_panics_doc)] // TODO
     #[instrument(name = "App::new", level = "debug")]
     pub async fn new(options: Options) -> EyreResult<Self> {
+        if let Ok(app) = Self::bootstrap(options.clone()).await {
+            Ok(app)
+        } else {
+            error!("Error when rebuilding tree from cache. Retrying with db cache busted.");
+
+            // Remove cached events from database and try again
+            let db = Database::new(options.database.clone()).await?;
+            db.wipe_cache().await?;
+
+            Self::bootstrap(options).await
+        }
+    }
+
+    async fn bootstrap(options: Options) -> EyreResult<Self> {
```

Comment:
Wrapping `new(..)` like this is a hack, we should instead make the wipe-and-retry part of the startup logic.